app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,39 @@
+class CommentsController < ApplicationController
+
+	def new
+		@post = Post.find(params[:post_id])
+		@comment = Comment.new
+	end
+
+	def create
+		@post = Post.find_by_id(params[:post_id])
+		@comment = @post.comments.create!(comm_params)
+
+		redirect_to post_path(@post)
+	end
+	def edit
+		@post = Post.find(params[:post_id])
+		@comment = @post.comments.find(params[:id])
+	end
+
+	def update
+		@post = Post.find(params[:post_id])
+		@comment = @post.comments.find(params[:id])
+		@comment.update(comm_params)
+
+		redirect_to post_path(@post)
+	end
+
+	def destroy
+		@post = Post.find(params[:post_id])
+		@comment = Comment.find(params[:id])
+		@comment.destroy
+
+		redirect_to post_path(@post)
+	end
+
+private
+	def comm_params
+		params.require(:comment).permit(:username,:text)
```

Comment:
If you're able to implement `current_user`, perhaps you wouldn't need to pass in the username via the form...